eclair-core/src/test/scala/fr/acinq/eclair/router/GraphSpec.scala

```diff
@@ -261,4 +262,89 @@ class GraphSpec extends AnyFunSuite {
       BlockHeight(714930), _ => true, includeLocalChannelCost = true)
     assert(path.path == Seq(edgeAB, edgeBC, edgeCE))
   }
+
+  test("fees less along C->D->E than C->E") {
+    /*
+    The channel C -> D is large enough for the payment and associated fees to go through, but C -> E is not.
+    A --> B --> C <-> D
+                 \   /
+                  \ /
+                   E
+    */
+    val edgeAB = makeEdge(1L, a, b, 10001 msat, 0, capacity = 200000 sat)
+    val edgeBC = makeEdge(2L, b, c, 10000 msat, 0, capacity = 200000 sat)
+    val edgeCD = makeEdge(3L, c, d, 10001 msat, 0, capacity = 200000 sat)
+    val edgeDC = makeEdge(4L, d, c, 10 msat, 0, capacity = 200000 sat)
+    val edgeCE = makeEdge(5L, c, e, 10003 msat, 0, capacity = 200000 sat)
```

Comment:
You wrote in the comment that "C -> D is large enough but C -> E is not" but C -> D and C -> E have the same capacity.

Reply:
I changed the comment to 
```
The channel going through C -> D -> E  (fee = 1001 + 1 = 1002) is considered shorter than going 
    through C -> E (fee = 1003) because the fees are less.
```